flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/FileStoreScan.java

```diff
@@ -56,5 +56,11 @@
 
         /** Result {@link ManifestEntry} files. */
         List<ManifestEntry> files();
+
+        /**
+         * Deleted sst files in current snapshot. They can be safely deleted from file system if
+         * this snapshot expires.
+         */
+        List<ManifestEntry> deletedFiles();
```

Comment:
This `deletedFiles` is misunderstanding.
I prefer to add a new method `planChangeFiles()`, it returns all files. The caller can compute physical deleted files.